packages/services/src/coin-selection/bitcoin-coin-selection.service.ts

```diff
@@ -0,0 +1,82 @@
+import { injectable } from 'inversify';
+
+import {
+  CalculateMaxSpendResponse,
+  CoinSelectionOutput,
+  CoinSelectionRecipient,
+  calculateMaxSpend,
+  determineUtxosForSpend,
+  determineUtxosForSpendAll,
+} from '@leather.io/bitcoin';
+import { Money, OwnedUtxo } from '@leather.io/models';
+
+import { LeatherApiClient } from '../infrastructure/api/leather/leather-api.client';
+import { AccountRequest } from '../types';
+import { UtxosService } from '../utxos/utxos.service';
+
+export interface CalculateMaxSpendRequest {
+  account: AccountRequest;
+  recipient: string;
+  feeRate?: number;
+}
+
+export interface CoinSelectionRequest {
+  account: AccountRequest;
+  recipients: CoinSelectionRecipient[];
+  feeRate?: number;
+  isMaxSpend?: boolean;
+}
+
+export interface CoinSelectionResult {
+  inputs: OwnedUtxo[];
+  outputs: CoinSelectionOutput[];
+  estimatedTxSize: number;
+  fee: Money;
+}
+
+@injectable()
+export class BitcoinCoinSelectionService {
+  constructor(
+    private readonly utxosService: UtxosService,
+    private readonly leatherApiClient: LeatherApiClient
+  ) {}
+
+  async performCoinSelection(
+    { account, recipients, feeRate, isMaxSpend }: CoinSelectionRequest,
+    signal?: AbortSignal
+  ): Promise<CoinSelectionResult> {
+    const { available: availableInputUtxos } = await this.utxosService.getAccountUtxos(
+      account,
+      signal
+    );
+    if (!feeRate) {
+      const feeRates = await this.leatherApiClient.fetchBitcoinFeeRates({ signal });
+      feeRate = feeRates.standard.rate;
+    }
+    const result = isMaxSpend
+      ? determineUtxosForSpendAll({ feeRate, recipients, utxos: availableInputUtxos })
```

Comment:
Maybe we can later rename `determineUtxosForSpendAll` as `determineMaxSpendUtxos`? 

It could be just me but reading this was unclear. Then when I looked into it I see we still have duplicate code in the `extension` and `package` for it we can refactor soon